### PR TITLE
implementation guidelines: logging

### PR DESCRIPTION
**Pull request by adwk67**

### CODE REVIEW

File: modules/contributor/pages/logging.adoc [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+= Logging aggregation implementation guidelines
+
+For a conceptual overview of logging aggregation, consult the xref:concepts:logging.adoc[logging concept page].
+
+== Overview
+
+=== Approach
+
+The Stackable Data Platform (SDP) does _*not*_ implement product logging where none exists, but rather provides a way for the user to interact with whatever logging framework is already in place, and to ensure that the log output can be consolidated and preserved in an output sink (e.g. Opensearch). Products use various logging libraries such as log4j, log4j2, logback, python logging etc. and SDP logging aggregation enables:
```
Comment (maltesander) [inline modules/contributor/pages/logging.adoc:12]: Just a suggestion: Id prefer capitalization here.
```suggestion
The Stackable Data Platform (SDP) does _*not*_ implement product logging where none exists, but rather provides a way for the user to interact with whatever logging framework is already in place, and to ensure that the log output can be consolidated and preserved in an output sink (e.g. Opensearch). Products use various logging libraries such as Log4j, Log4j2, Logback, Python logging etc. and SDP logging aggregation enables:
```

File: modules/contributor/pages/logging.adoc [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+
+- the extension or overriding of the default product logging configuration (so that e.g. the log level can be changed): this assumes knowledge of the product logging library
+- the capture and parsing of stdout and stderr logging statements
+- the consolidation of both of the above into a standard logging format which will be written to the aggregation sink
+
+Implementing logging for a given product will require code:
```
Comment (maltesander) [inline modules/contributor/pages/logging.adoc:18]:
```suggestion
Implementing logging for a given operator and product will require code:
```

File: modules/contributor/pages/logging.adoc [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+[source,rust]
+----
+pub fn extend_config_map<C, K>(
+    role_group: &RoleGroupRef<K>,
+    vector_aggregator_address: Option<&str>,
+    logging: &Logging<C>,
+    ... // other parameters dependent on calling context
+    cm_builder: &mut ConfigMapBuilder,
+) -> Result<()>
```
Comment (maltesander) [inline modules/contributor/pages/logging.adoc:54]: The example above had an opening "{". I would keep this consistent.

File: modules/contributor/pages/logging.adoc [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+- a default logging configuration for the given product logging library e.g. `product_logging::framework::create_log4j2_config`
+- a Vector configuration called from `product_logging::framework::create_vector_config`
+
+For the first of these, the framework provides pre-configured configurations for several familiar libraries. These can be overridden by providing a specific logging configuration in a `ConfigMap` directly in the custom resource. The second one is set by a boolean flag. Both these are shown in the snippet below:
+
+[source,yaml]
+----
+    config:
+      logging:
+        enableVectorAgent: true # indicates vector.toml should be mounted
```
Comment (maltesander) [inline modules/contributor/pages/logging.adoc:68]: This also indicates whether or not we will deploy a vector container with the log agent.

File: modules/contributor/pages/logging.adoc [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+        r#"handlers=java.util.logging.FileHandler, java.util.logging.ConsoleHandler
+.level={root_log_level}
+java.util.logging.FileHandler.pattern=/path/to/jul.%g.logger
+java.util.logging.FileHandler.limit=50000
+java.util.logging.FileHandler.count=10
+java.util.logging.FileHandler.formatter=java.util.logging.SimpleFormatter
+java.util.logging.ConsoleHandler.level={console_log_level}
+java.util.logging.ConsoleHandler.formatter = java.util.logging.SimpleFormatter
+
+"#,
```
Comment (maltesander) [inline modules/contributor/pages/logging.adoc:91]: I **think** this should look better in the preview.
```suggestion
fn create_java_logging_config(
) -> String {
    format!(r#"
handlers=java.util.logging.FileHandler, java.util.logging.ConsoleHandler
.level={root_log_level}
java.util.logging.FileHandler.pattern=/path/to/jul.%g.logger
java.util.logging.FileHandler.limit=50000
java.util.logging.FileHandler.count=10
java.util.logging.FileHandler.formatter=java.util.logging.SimpleFormatter
java.util.logging.ConsoleHandler.level={console_log_level}
java.util.logging.ConsoleHandler.formatter = java.util.logging.SimpleFormatter

"#,
```

**Review by maltesander — CHANGES_REQUESTED** [review]
Just minor things. LGTM overall!

File: modules/contributor/pages/logging.adoc (hunk 3 above)
Comment (adwk67) [reply modules/contributor/pages/logging.adoc]: Done.

File: modules/contributor/pages/logging.adoc [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+- a default logging configuration for the given product logging library e.g. `product_logging::framework::create_log4j2_config`
+- a Vector configuration called from `product_logging::framework::create_vector_config`
+
+For the first of these, the framework provides pre-configured configurations for several familiar libraries. These can be overridden by providing a specific logging configuration in a `ConfigMap` directly in the custom resource. The second one is set by a boolean flag. Both these are shown in the snippet below:
+
+[source,yaml]
+----
+    config:
+      logging:
+        enableVectorAgent: true # deploy a vector container with the log agent and  mount vector.toml
```
Comment (maltesander) [inline modules/contributor/pages/logging.adoc:68]:
```suggestion
        enableVectorAgent: true # deploy a vector container with the log agent and mount the vector.toml config
```

**Review by maltesander — DISMISSED** [review]
LGTM!

**Review by maltesander — APPROVED** [review]

**Pull request merged**